charts/catalog/templates/apiserver-deployment.yaml

```diff
@@ -20,6 +20,7 @@ spec:
         release: "{{ .Release.Name }}"
         heritage: "{{ .Release.Service }}"
     spec:
+      serviceAccountName: server
```

Comment:
I think this and the `serviceAccountName` in `controller-manager-deployment.yaml` would be worth parameterizing in the `values.yaml` file as well